.test/run.sh

```diff
@@ -92,6 +92,97 @@ mariadbclient_unix() {
 		"$@"
 }
 
+checkUserExistInMariaDB() {
+	if [ -z $1 ] ; then
+		return 1
+	fi
+
+	local user=$(mariadbclient -u root -e "SELECT User FROM mysql.user where User='$1';")
+	if [ -z $user ] ; then
+		return 1
+	fi
+
+	return 0
+}
+
+checkReplication() {
+	mariadb_replication_user='foo'
+	local pass_str=
+	local pass=
+	if [ $1 = 'MARIADB_REPLICATION_PASSWORD_HASH' ] ; then 
+		pass_str=MARIADB_REPLICATION_PASSWORD_HASH='*0FD9A3F0F816D076CF239580A68A1147C250EB7B'
+		pass='jane'
+	else
+		pass_str='MARIADB_REPLICATION_PASSWORD=foo123'
+		pass='foo123'
+	fi
+
+	runandwait \
+		--publish 3306 \
+		-e MARIADB_REPLICATION_MODE="master" \
+		-e MARIADB_REPLICATION_USER="$mariadb_replication_user" \
+		-e $pass_str \
+		-e MARIADB_DATABASE=replcheck \
+		-e MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=1 \
+		-e MARIADB_MYSQL_LOCALHOST_USER=1 \
+		-e MARIADB_MYSQL_LOCALHOST_GRANTS="REPLICA MONITOR" \
+		"$image" --server-id=3000 --log-bin --log-basename=my-mariadb
+
+	# Checks $mariadb_replication_user get created or not
+	if checkUserExistInMariaDB $mariadb_replication_user ; then
+		grants=$(mariadbclient -u $mariadb_replication_user -p$pass -e "SHOW GRANTS")
+		[[ "${grants/SLAVE/REPLICA}" =~ "GRANT REPLICATION REPLICA ON *.* TO \`$mariadb_replication_user\`@\`%\`" ]] || die "I wasn't created how I was expected: got $grants"
+
+		mariadbclient -u root --batch --skip-column-names -e 'create table t1(i int)' replcheck
+		readarray -t vals < <(mariadbclient -u root --batch --skip-column-names -e 'show master status\G' replcheck)
+		lastfile="${vals[1]}"
+		pos="${vals[2]}"
+		[[ "$lastfile" = my-mariadb-bin.00000[12] ]] || die "too many binlog files"
+		[ "$pos" -lt 500 ] || die 'binary log too big'
+		docker exec "$cid" ls -la /var/lib/mysql/my-mariadb-bin.000001
+		docker exec "$cid" sh -c '[ $(wc -c < /var/lib/mysql/my-mariadb-bin.000001 ) -gt 2500 ]' && die 'binary log 1 too big'
+		docker exec "$cid" sh -c "[ \$(wc -c < /var/lib/mysql/$lastfile ) -gt $pos ]" && die 'binary log 2 too big'
+
+		master_host=$cname
+		master_cid=$cid
+		master_host_ip=$(docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $master_cid)
+		cname="mariadb-container-$RANDOM-$RANDOM"
+		cid=$(docker run \
```

Comment:
copy the simplification from dc1ac13b54c1005a28254e840023bd5bc12b3b17

Reply:
Copied